### ISSUE TITLE
Performance issues on refresh()

### ISSUE THREAD

**Issue by leo-combes** [SitePen/dgrid#1115]
I'm using dStore with dgrid this way:

1- First, fill a Memory dStore with 170 items.

storeDemo = new Memory({data: data, idProperty: 'index'});

2- Then I assign the store to the collection of a dGrid.

gridDemo.set('collection', storeDemo);

3- Later, through a forEach, I walk all items in the store one by one by one, I update its value to "waiting" in store, call gridDemo.refresh(), and then try to update its row value by XHR request to then assign the new value in the store.
Finally refresh the grid again.

	storeDemo.filter({nas_enabled: "1"}).forEach(function (entry) {
		update_status(entry.index);
	});

	function update_status(mac_address){

		var item = storeDemo.getSync(mac_address);

		gridDemo.row(item.index).data.col_1 = "wait";
		gridDemo.row(item.index).data.col_2 = "wait";
		gridDemo.row(item.index).data.col_3 = "wait";	

		gridDemo.refresh();

		var deferred = load_new_info(put_data, 10000);

		deferred.then(function(result){
			item.col_1 = result.a;
			item.col_2 = result.b;
			item.col_3 = result.c;

		    storeDemo.put(item).then(function(){
		    	gridDemo.refresh();
		});	
    }

This operation, while it is running, makes firefox eat 60% - 80% of CPU.
If I try update 2 or 3 rows at same time, CPU usage raises at 100% all the time.

If I make the whole process in the same way, but only removing calls to refresh(), this is, working only with the store, firefox uses only 5-15% of CPU.

NOTE: forEach the code is simplified, but we make sure that only one call to update_status () is done at the same time. Each call must finish before doing another.

In total there are 2-4 calls per second to update_status() until the entire table is updated.
there any way to do this with better performance? Or is normal to expect that consumption of resources?

Linux 3.13 x86_64 with Firefox 37
